app/Jobs/WikiEntityImportJob.php

```diff
@@ -0,0 +1,250 @@
+<?php
+
+namespace App\Jobs;
+
+use App\WikiEntityImportStatus;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Support\Facades\Http;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Support\Facades\Config;
+use App\Wiki;
+use App\WikiEntityImport;
+use Carbon\Carbon;
+use Maclof\Kubernetes\Client;
+use Maclof\Kubernetes\Models\Job as KubernetesJob;
+
+class WikiEntityImportJob implements ShouldQueue
+{
+    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
+
+    /**
+     * Create a new job instance.
+     */
+    public function __construct(
+        public int $wikiId,
+        public string $sourceWikiUrl,
+        public array $entityIds,
+        public int $importId,
+    )
+    {}
+
+    private string $targetWikiUrl;
+
+    /**
+     * Execute the job.
+     */
+    public function handle(Client $kubernetesClient): void
+    {
+        $import = null;
+        try {
+            $wiki = Wiki::findOrFail($this->wikiId);
+            $import = WikiEntityImport::findOrFail($this->importId);
+            $creds = $this->acquireCredentials($wiki->domain);
+
+            $this->targetWikiUrl = $this->domainToOrigin($wiki->domain);
+
+            $kubernetesJob = new TransferBotKubernetesJob(
+                kubernetesClient: $kubernetesClient,
+                wiki: $wiki,
+                creds: $creds,
+                entityIds: $this->entityIds,
+                sourceWikiUrl: $this->sourceWikiUrl,
+                targetWikiUrl: $this->targetWikiUrl,
+                importId: $this->importId,
+            );
+            $jobName = $kubernetesJob->spawn();
+            Log::info(
+                'transferbot job for wiki "'.$wiki->domain.'" was created with name "'.$jobName.'".'
+            );
+        } catch (\Exception $ex) {
+            Log::error('Entity import job failed with error: '.$ex->getMessage());
+            $import?->update([
+                'status' => WikiEntityImportStatus::Failed,
+                'finished_at' => Carbon::now(),
+            ]);
+            $this->fail(
+                new \Exception('Error spawning transferbot for wiki '.$this->wikiId.': '.$ex->getMessage()),
+            );
+        }
+    }
+
+    private static function domainToOrigin(string $domain): string
+    {
+        $tld = last(explode('.', $domain));
+        return $tld === 'localhost'
+            ? "http://".$domain
+            : "https://".$domain;
+    }
+
+    private static function acquireCredentials(string $wikiDomain): OAuthCredentials
```

Comment:
comment: after the testing thoughts yesterday I was left wondering if I should be recommending moving this logic into the transferbot image but I think that it probably still makes most sense here. This is slightly different to QuickStatements et al. but given that all the import stuff is now split between here and the `transferbot` image I think it's a fair argument to keep the content of the transferbot image fairly minimal

Reply:
It would also mean the transferbot image would be super tightly bound to wbstack and can't even be used in any other setup. It would be a thing we could do, but if we do it, we might also de-generalize other parts in there (i.e. selecting wikis by Host header and such).